patches/server/0964-Properly-handle-BlockBreakEvent-isDropItems.patch

```diff
@@ -0,0 +1,106 @@
+-                        block.playerDestroy(this.level, this.player, pos, iblockdata, tileentity, itemstack1);
++                    if (flag && flag1 /* && event.isDropItems() */) { // CraftBukkit - Check if block should drop items // Paper - fix drops not preventing stats/food exhaustion
++                        block.playerDestroy(this.level, this.player, pos, iblockdata, tileentity, itemstack1, event.isDropItems()); // Paper
+                     }
+ 
+                     // return true; // CraftBukkit
+diff --git a/src/main/java/net/minecraft/world/level/block/BeehiveBlock.java b/src/main/java/net/minecraft/world/level/block/BeehiveBlock.java
+index fe7daa57119af137eeb76ca2c0e54bf7903ebbca..71694d7d76ec9bd67e9ec7408df567505844c947 100644
+--- a/src/main/java/net/minecraft/world/level/block/BeehiveBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/BeehiveBlock.java
+@@ -77,8 +77,8 @@ public class BeehiveBlock extends BaseEntityBlock {
+     }
+ 
+     @Override
+-    public void playerDestroy(Level world, Player player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, ItemStack stack) {
+-        super.playerDestroy(world, player, pos, state, blockEntity, stack);
++    public void playerDestroy(Level world, Player player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, ItemStack stack, boolean includeDrops) { // Paper
++        super.playerDestroy(world, player, pos, state, blockEntity, stack, includeDrops); // Paper
+         if (!world.isClientSide && blockEntity instanceof BeehiveBlockEntity) {
+             BeehiveBlockEntity tileentitybeehive = (BeehiveBlockEntity) blockEntity;
+ 
+diff --git a/src/main/java/net/minecraft/world/level/block/Block.java b/src/main/java/net/minecraft/world/level/block/Block.java
+index 7b71073027f4cf79736546500ededdfbb83d968e..e3fb9a532d1f7c217ac99cdfd316f822a92289c7 100644
+--- a/src/main/java/net/minecraft/world/level/block/Block.java
++++ b/src/main/java/net/minecraft/world/level/block/Block.java
+@@ -432,9 +432,16 @@ public class Block extends BlockBehaviour implements ItemLike {
+         return this.defaultBlockState();
+     }
+ 
++    @io.papermc.paper.annotation.DoNotUse // Paper - method below allows better control of item drops
+     public void playerDestroy(Level world, Player player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, ItemStack stack) {
++        // Paper start
++        this.playerDestroy(world, player, pos, state, blockEntity, stack, true);
++    }
++    public void playerDestroy(Level world, Player player, BlockPos pos, BlockState state, @Nullable BlockEntity blockEntity, ItemStack stack, boolean includeDrops) {
++        // Paper end
+         player.awardStat(Stats.BLOCK_MINED.get(this));
+         player.causeFoodExhaustion(0.005F, org.bukkit.event.entity.EntityExhaustionEvent.ExhaustionReason.BLOCK_MINED); // CraftBukkit - EntityExhaustionEvent
++        if (includeDrops) // Paper
```

Comment:
Wouldn't we want to include the call to `Block.dropResources` in the diff here to ensure failure on change ?
If statement without brackets just seems fun :sweat_smile:

Reply:
the diff is close enough to confligt. If the line below changes, the diff will conflict because when it inserts the line above it, it will expect the line below it to be what it is now.

Reply:
Okay yea, wasn't sure specifically when it breaks :+1: